satellite/gc/gc_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gc_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/internal/testrand"
+	"storj.io/storj/pkg/encryption"
+	"storj.io/storj/pkg/paths"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+)
+
+// TestGarbageCollection does the following:
+// * Set up a network with one storagenode
+// * Upload two objects
+// * Delete one object from the metainfo service on the satellite
+// * Wait for bloom filter generation
+// * Check that pieces of the deleted object are deleted on the storagenode
+// * Check that pieces of the kept object are not deleted on the storagenode
+func TestGarbageCollection(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 1, UplinkCount: 1,
+		Reconfigure: testplanet.Reconfigure{
+			Satellite: func(log *zap.Logger, index int, config *satellite.Config) {
+				config.GarbageCollection.FalsePositiveRate = 0.001
```

Comment:
Using `0.001` means that once in 1000 times we will have a false positive and this test will fail, right? 

Can we use the smallest possible number here?

Reply:
In theory, yeah. The smaller the number, the larger the bloom filter I think. I don't know what a reasonable value is, but the test has not failed as a result of this yet.